js/index8.js

```diff
@@ -548,15 +548,647 @@ console.log(uniqueSortedCourses); // ['biologie', 'informatică', 'limba englez
+console.log(`-Example 2 - Map method-`);
+
+const makeCarsWithDiscount = (cars, discount) => {
+  return cars.map((x) => x.price - discount * x.price);
+};
+
+console.table(makeCarsWithDiscount(cars, 0.2));
+console.table(makeCarsWithDiscount(cars, 0.4));
+
+/*
+Example 3 -Filter method
+Let the `filterByPrice` function return an array of cars whose price is less
+than the value of the `threshold` parameter.
+*/
+
+console.log(`-Example 3 - Filter method-`);
+
+const filterByPrice = (cars, threshold) => {
+  return cars.filter((x) => x.price < threshold);
+};
+
+console.table(filterByPrice(cars, 30000));
+console.table(filterByPrice(cars, 25000));
+
+/*
+Example 4 - Filter method
+Let the `getCarsWithDiscount` function return an array of cars whose
+onSale property is true.
+*/
+
+console.log(`-Example 4 - Filter method-`);
+
+const getCarsWithDiscount = (cars) => {
+  return cars.filter((x) => x.onSale === true);
+};
+
+console.table(getCarsWithDiscount(cars));
+
+/*
+Example 5 - Filter method
+Let the `getCarsWithType` function return an array of cars whose type
+coincide with the value of the `type` parameter.
+*/
+
+console.log(`-Example 5 - Filter method-`);
+
+const getCarsWithType = (cars, type) => {
+  return cars.filter((x) => x.type === type);
+};
+
+console.table(getCarsWithType(cars, "suv"));
+console.table(getCarsWithType(cars, "sedan"));
+
+/*
+Example 6 - The find method
+*/
+
+console.log(`-Example 6 - The find method
+-`);
+
+const getCarByModel = (cars, model) => {
+  return cars.find((x) => x.model === model);
+};
+
+console.log(getCarByModel(cars, "F-150"));
+console.log(getCarByModel(cars, "CX-9"));
+
+/*
+Example 7 - Sort method
+Let the `sortByAscendingAmount` function return a new array of cars
+sorted in ascending order by the value of the `amount` property.
+*/
+
+console.log(`-Example 7 - Sort method
+-`);
+
+const sortByAscendingAmount = (cars) => {
+  return [...cars].sort((x, y) => x.amount - y.amount);
+};
+
+console.table(sortByAscendingAmount(cars));
+
+/*
+Example 8 - Sort method
+Let the `sortByDescendingPrice` function return a new array of cars
+sorted in descending order by the value of the `price` property.
+*/
+
+console.log(`-Example 8 - Sort method
+-`);
+
+const sortByDescendingPrice = (cars) => {
+  return [...cars].sort((x, y) => y.price - x.price);
+};
+
+console.table(sortByDescendingPrice(cars));
+
+/*
+Example 9 - Sort method
+Let the `sortByModel` function return a new array of cars sorted
+by model name in alphabetical and reverse alphabetical order, depending on
+the value of the `order` parameter.
+*/
+
+console.log(`-Example 9 - Sort method
+-`);
+
+const sortByModel = (cars, order) => {
+  // return a.sortValue?.[1].localeCompare(b.sortValue?.[1]);
+  let newCars;
+  if (order === "asc") {
+    newCars = [...cars].sort((x, y) => x.make.localeCompare(y.make));
+  } else if (order === "desc") {
+    newCars = [...cars].sort((x, y) => y.make.localeCompare(x.make));
+  }
+
+  return newCars;
+};
```

Comment:
Nice!

Reply:
One small thing: you should handle the case when order is not one of "asc" or "desc"